src/parser/flags.ts

```diff
@@ -42,6 +43,20 @@ export const integer = build({
   },
 })
 
+export const directory = (opts: { exists?: boolean } & Partial<OptionFlag<string>> = {}): OptionFlag<string | undefined> => {
+  return build<string>({
+    ...opts,
+    parse: async (input: string) => opts.exists ? dirExists(input) : input,
+  })()
+}
+
+export const file = (opts: { exists?: boolean } & Partial<OptionFlag<string>> = {}): OptionFlag<string | undefined> => {
+  return build<string>({
+    ...opts,
+    parse: async (input: string) => opts.exists ? fileExists(input) : input,
```

Comment:
If `exists` is `false` this should at least validate the input is a valid file path.